### ISSUE TITLE
Add Eclipse-SourceReferences during build

### ISSUE THREAD

**Issue by kthoms** [eclipse/xtext#1504]
The SimRel tests reveal in [esdata.txt](https://ci-staging.eclipse.org/xtext/job/releng/job/simrel-tests/lastSuccessfulBuild/artifact/build-result/reporeports/reports/esdata.txt) that Xtext bundles do not have a `Eclipse-SourceReferences` manifest header.

For Tycho builds there is an instruction how to add this here: https://wiki.eclipse.org/PDE/UI/SourceReferences#Generating_Source_Reference_Headers_with_Tycho

For bundles built with Gradle this has to be achieved differently.

**Comment by cdietrich** [comment]
https://bugs.eclipse.org/bugs/show_bug.cgi?id=426430

**Comment by cdietrich** [comment]
possible solution for gradle

adapt manifests to contain a placeholder

```
Eclipse-SourceReferences: eclipseSourceReferences
```
TODO: check what def eclipse thinks about that

adapt `manifest-gen.gradle` to calc the stuff

```
manifestReplacements.put("eclipseSourceReferences", eclipseSourceReferences())
...
def eclipseSourceReferences() {
	def rootPath = java.nio.file.Paths.get(project.rootProject.projectDir.canonicalPath)
	def projectPath = java.nio.file.Paths.get(project.projectDir.canonicalPath)
	def relativePath = rootPath.relativize(projectPath)
	def url = grgit.remote.list().first().url
	def commitId = grgit.head().id
	return "scm:git:${url};path=\"${relativePath}\"" + (commitId ? ";commitId=${commitId}" : "")
}
```

**Comment by cdietrich** [comment]
whilst we are at maybe we should move the version of the grgit plugin to the versions.gradle too

**Comment by cdietrich** [comment]
the main problem is that gradle does not really like single projects so the project will not be really usable when imported into a eclipse (without root projects etc)

for testing: https://ci.eclipse.org/xtext/job/xtext-umbrella/job/cd_xtext_issue1504/lastSuccessfulBuild/artifact/build/p2-repository/

**Comment by cdietrich** [comment]
fixed in 2.20
